sopel/coretasks.py

```diff
@@ -94,8 +98,10 @@ def _handle_sasl_capability(
     # Manage CAP REQ :sasl
     auth_method = bot.settings.core.auth_method
     server_auth_method = bot.settings.core.server_auth_method
+    auth_target = bot.settings.core.auth_target
```

Comment:
If only `server_auth_method` is set, then you need to check `server_auth_sasl_mech`, instead of `auth_target`.

Reply:
Fixed. I wouldn't mind making these config options less confusing at some point...